# Intervalos de confianza

In [1]:
import numpy as np
import matplotlib.pyplot as plt
import scipy.stats as st

## Intervalo de Confianza para la media de una distribución normal (σ conocida)
Sea $X_1, \dots, X_n$ una m.a. $X_i \sim \mathcal{N}(\mu, \sigma^2)$. Entonces:
- $\bar{X} \sim \mathcal{N}(\mu, \frac{\sigma^2}{n})$
- $Z = \frac{\bar{X} - \mu}{\sigma / \sqrt{n}} \sim \mathcal{N}(0,1)$
### Nivel de significancia (error): $\alpha$
El intervalo de confianza se basa en que:
$$
\mathbb{P}\left( -z_{\alpha/2} < \frac{\bar{X} - \mu}{\sigma / \sqrt{n}} < z_{\alpha/2} \right)
= \mathbb{P}\left( \bar{X} - z_{\alpha/2} \frac{\sigma}{\sqrt{n}} < \mu < \bar{X} + z_{\alpha/2} \frac{\sigma}{\sqrt{n}} \right)
= 1 - \alpha
$$
### El intervalo del $(1 - \alpha) \cdot 100\%$ de confianza para $\mu$ (con $\sigma$ conocida) es:
$$
\left( \bar{X} - z_{\alpha/2} \frac{\sigma}{\sqrt{n}}, \quad \bar{X} + z_{\alpha/2} \frac{\sigma}{\sqrt{n}} \right)
$$
donde
 - $\sigma$ es la desviación estándar
 - $\frac{\sigma}{\sqrt{n}}$ es el error estándar de la media 
 - $z_{\alpha/2}$ determina el nivel de confianza
 - el intervalo está centrado en $\bar{X}$

 **OBS:** A mayor valor de $n \rightarrow$ menor error estándar $\rightarrow$ el intevalo de confianza es más pequeño $\rightarrow$ lo cual hace que se acerque más a la media, es decir, que haya menos variación.

**Ejemplo 1**

Para tratar de estimar la media de consumo por cliente de un restaurante, se reunieron datos de una muestra de $49$ clientes durante tres semanas.

**a)** Supongamos que la **desviación estándar** de la población es de $\$2.5$. ¿Cuál es el error estándar de la media?

**b)** Con un nivel de confianza del $95\%$, ¿cuál es el margen de error?

**c)** Si la media de la muestra es de $\$22.6$, ¿cuál es el intervalo de confianza del $95\%$ para la media de la población?

In [4]:
# Datos
n = 49
sigma = 2.5
media_muestral = 22.6
confianza = 0.95

# a) Error estandar de la media 

error_std = sigma/np.sqrt(n)
print(f'Error estándar de la media: {error_std:.2f}')


# b) Margen de error 
# 1-confianza = 0.05 => 0.025 en cada cola => (1-confianza)/2

z = st.norm.ppf(1 - (1-confianza)/2) # ppf -> percentil point function 
# z_{0.975} es tal que P(Z <= z_{0.975}) = 0.975
margen_err = z * error_std

print(f'Margen de error: {margen_err:.2f}')


# c) Intervalo de confianza
lim_sup = media_muestral + margen_err
lim_inf = media_muestral - margen_err

print(f'Intervalo de confianza del {confianza*100}%: ({lim_inf:.2f},{lim_sup:.2f})')

Error estándar de la media: 0.36
Margen de error: 0.70
Intervalo de confianza del 95.0%: (21.90,23.30)


Por lo tanto, podemos decir que con un $95\%$ de confianza, se estima que el **consumo medio poblacional** está entre $\$21.90$ y $\$23.30$

**Ejemplo 2** 

Supongamos que se toma una muestra aleatoria de 100 personas para estimar la media del peso de una población, y se obtiene que la media muestral es de $70$ kg con una desviación estándar que es conocida de $10$ kg. Para un nivel de confianza del $95\%$, calcular el intervalo de confianza

In [7]:
# Datos
n = 100
sigma = 10
media_muestral = 70
confianza = 0.95

error_std = sigma/np.sqrt(n)

'''
Vamos a utilizar la funcion scipy.stats.norm.interval

Sintaxis: scipy.stats.norm.interval(confidence=confianza, loc=media_muestral, scale=error_std)
'''

intervalo = st.norm.interval(confianza,loc=media_muestral, scale=error_std)

print(f'El intervalo de confianza del {confianza*100}% es: ({intervalo[0]:.2f},{intervalo[1]:.2f})')

El intervalo de confianza del 95.0% es: (68.04,71.96)


## Intervalo de Confianza para la media $\mu$ de una distribución normal ($\sigma$ deconocida)
Sea $X_1, \dots, X_n$ una m.a. $X_i \sim \mathcal{N}(\mu, \sigma^2)$. Entonces:
- $\bar{X} \sim \mathcal{N}(\mu, \frac{\sigma^2}{n})$
- $Z = \frac{\bar{X} - \mu}{\sigma / \sqrt{n}} \sim \mathcal{N}(0,1)$
Si el tamaño de la muestra es menor a 30, se utiliza la $t$: 
### Estadístico t
Definimos el estadístico $t$ de la siguiente manera:
$$T = \frac{\bar{X}-\mu}{s/\sqrt{n}} \sim t_{(n-1)} $$
### El intervalo del $(1 - \alpha) \cdot 100\%$ de confianza para $\mu$ (con $\sigma$ desconocida) es:
$$
\left( \bar{X} - t_{\alpha/2} \frac{s}{\sqrt{n}}, \quad \bar{X} + t_{\alpha/2} \frac{s}{\sqrt{n}} \right)
$$
donde
- $s$ = desviación estándar muestral
- $\frac{s}{\sqrt{n}}$ = error estándar de la media  
- $t_{\alpha/2}$ determina el nivel de confianza  
- El intervalo está centrado en $\bar{X}$

**Ejemplo 3** 

Supongamos que tenemos los siguientes datos de la muestra aleatoria de 10 observaciones:
$$ datos = \{45,55,67,45,68,79,98,87,84,82\}$$

Calcular un intervalo de confianza para la media.

In [9]:
# Datos
datos = [45,55,67,45,68,79,98,87,84,82]
n = len(datos)
confianza = 0.95
df = n - 1

# Tenemos la media y el error estandar
media_muestral = np.mean(datos)

error_std = st.sem(datos) # std error of the mean utilizando la desviación estándar muestral

# Utilizamos la distribución t de Student
intervalo = st.t.interval(confianza, df=df, loc=media_muestral, scale=error_std)

print(f'Media muestral es: {media_muestral:.2f}')
print(f'El intervalo de confianza del {confianza*100}% es: ({intervalo[0]:.4f},{intervalo[1]:.4f})')

Media muestral es: 71.00
El intervalo de confianza del 95.0% es: (58.0005,83.9995)


**Ejemplo 4** 

Los artículos de cerámica utilizados sobre velas electricas sobrecargadas se rompen con diferentes presiones. Supongamos que los datos provienen de una distribución normal.
La resistencia a la ruptura fue medida en una muestra de 100 artículos, y el promedio fue de $1750$ con un desviación estándar de 315.8

**a)** Estimar con un nivel del confianza del $90\%$ a la media poblacional de la presión de la ruptura.

**b)** Estimar con un nivel del confianza del $90\%$ a la varianza poblacional.

In [10]:
# Datos
n = 100
media_muestal = 1750
desv_muestral = 315.8
confianza = 0.90

# a) Se tiene que n>30, por lo tanto usamos la normal
error_std = desv_muestral/np.sqrt(n)
intervalo = st.norm.interval(confianza, loc=media_muestal, scale=error_std)
print(f'El intervalo de confianza del {confianza*100}% es: ({intervalo[0]:.2f},{intervalo[1]:.2f})')


El intervalo de confianza del 90.0% es: (1698.06,1801.94)


In [16]:
# b) Intervalo de confianza para la varianza
gl = n - 1
alpha = 1 - confianza
s2 = desv_muestral**2

# Vamos a calcular los cuantiles de la distribución chi-cuadrado
chi2_inf = st.chi2.ppf(alpha/2,df=gl)
chi2_sup = st.chi2.ppf(1 - alpha/2, df=gl)
inter_vari = ((gl*s2)/chi2_sup,(gl*s2)/chi2_inf)
print(f'El intervalo de confianza para la varianza del {confianza*100}% es: ({inter_vari[0]:.2f},{inter_vari[1]:.2f})')

El intervalo de confianza para la varianza del 90.0% es: (80123.49,128146.72)


In [17]:
# Otra forma de calcular el intervalo de confianza para la varianza
chi2_inf,chi2_sup = st.chi2.interval(confianza,df=gl)
inter_vari2 = ((gl*s2)/chi2_sup,(gl*s2)/chi2_inf)
print(f'El intervalo de confianza para la varianza del {confianza*100}% es: ({inter_vari2[0]:.2f},{inter_vari2[1]:.2f})')

El intervalo de confianza para la varianza del 90.0% es: (80123.49,128146.72)
